prov/efa/test/efa_unit_test_hmem.c

```diff
@@ -0,0 +1,48 @@
+#include "efa_unit_tests.h"
+
+
+#if HAVE_NEURON
+/**
+ * @brief Verify when neuron_alloc failed (return null),
+ * efa_domain_open, which call efa_hmem_support_status_update_neuron
+ * when HAVE_NEURON=1, will still return 0 but leave 
+ * hmem_support_status[FI_HMEM_NEURON].initialized and 
+ * hmem_support_status[FI_HMEM_NEURON].p2p_supported as false.
+ */
+void test_efa_hmem_support_status_update_neuron()
+{
+        int ret;
+        struct efa_resource resource = {0};
+        struct efa_domain *efa_domain;
+
+        resource.hints = efa_unit_test_alloc_hints(FI_EP_RDM);
+        assert_non_null(resource.hints);
+
+        ret = fi_getinfo(FI_VERSION(1, 14), NULL, NULL, 0ULL, resource.hints, &resource.info);
+        assert_int_equal(ret, 0);
+
+        ret = fi_fabric(resource.info->fabric_attr, &resource.fabric, NULL);
+        assert_int_equal(ret, 0);
+
+        hmem_ops[FI_HMEM_NEURON].initialized = true;
```

Comment:
Can you explain why you need to set this?

Reply:
This is to ensure `ofi_hmem_is_initialized()` https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/efa_hmem.c#L110-L122 return true so `efa_hmem_support_status_update_neuron` can reach the `neuron_alloc` line.

Reply:
It seems this value need to be recovered after test too. Sorry I missed this earlier.

Reply:
It was resetting it to `false`. But I changed it to save the original value and then recover to orig value, in case the original value is `true`.